clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1709,6 +1715,10 @@ private void updateLastSeenEpochIfNewer(TopicPartition topicPartition, OffsetAnd
             offsetAndMetadata.leaderEpoch().ifPresent(epoch -> metadata.updateLastSeenEpochIfNewer(topicPartition, epoch));
     }
 
+    private void sendFetches(Timer timer) {
+        applicationEventHandler.addAndGet(new FetchEvent(calculateDeadlineMs(timer)));
```

Comment:
this new event is really about `SendFetch` (not actual fetching), so maybe it makes the intention clearer if we rename it (aligned with the sendFetches func that creates it. (maybe me but reading FetchEvent here makes me go to the AppEventProcessor to make sure that we're not fetching/blocking there :) and it's only for sending the request)

Reply:
I started off renaming it as `SendFetch` but then that didn't seem right because the event doesn't directly send anything, either. Then I named it `GenerateFetchRequestsEvents`, which is... not good. I settled on `CreateFetchRequestsEvents` because that's exactly what it does.

I am not _too_ worried about the name, so if you really prefer `SendFetch` I can keep it as that.

Thanks.